packages/server/src/course/course.service.ts

```diff
@@ -206,28 +208,69 @@ export class CourseService {
         role: Role.PROFESSOR,
       }).save();
 
-      try {
-        // Update professor's last registered semester to semester model's current semester
-        let profLastRegistered: LastRegistrationModel;
-        profLastRegistered = await LastRegistrationModel.findOne({
-          where: { profId: userId },
-        });
-        if (profLastRegistered) {
-          profLastRegistered.lastRegisteredSemester = sectionGroup.semester;
-          await profLastRegistered.save();
-        } else {
-          profLastRegistered = await LastRegistrationModel.create({
-            profId: userId,
-            lastRegisteredSemester: sectionGroup.semester,
-          }).save();
-        }
-      } catch (err) {
-        console.error(err);
-        throw new HttpException(
-          ERROR_MESSAGES.courseController.updateProfLastRegistered,
-          HttpStatus.INTERNAL_SERVER_ERROR,
-        );
+      lastRegisteredSemester = sectionGroup.semester;
+    }
+
+    try {
+      // Update professor's last registered semester to semester model's current semester
+      let profLastRegistered: LastRegistrationModel;
+      profLastRegistered = await LastRegistrationModel.findOne({
+        where: { profId: userId },
+      });
+
+      // if the professor did not register anything, then find the current semester
+      if (!profLastRegistered) {
+        lastRegisteredSemester = this.getCurrentKhourySemester();
       }
+
+      if (profLastRegistered) {
+        profLastRegistered.lastRegisteredSemester = lastRegisteredSemester;
+        await profLastRegistered.save();
+      } else {
+        profLastRegistered = await LastRegistrationModel.create({
+          profId: userId,
+          lastRegisteredSemester,
+        }).save();
+      }
+    } catch (err) {
+      console.error(err);
+      throw new HttpException(
+        ERROR_MESSAGES.courseController.updateProfLastRegistered,
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  //TODO: harden course registration for Summer Courses. Last Registered Semester can't tell between summer 1 and summer full.
+  getCurrentKhourySemester = (): string => {
+    const now = new Date();
+    let year = now.getFullYear();
+    const season = this.monthToSeason(now.getMonth());
+
+    const courseSeasonMap = {
+      Fall: '10',
+      Spring: '30',
+      Summer_1: '40',
+      Summer_Full: '50',
+      Summer_2: '60',
+    };
+
+    if (season === 'Fall') {
+      year -= 1;
+    }
+
+    return `${year}${courseSeasonMap[season]}`;
+  };
+
+  private monthToSeason(month: number): string {
```

Comment:
I think this *could* be problematic in cases where like the start of the semester is the beginning of a month, right? (e.g. May 1st is the start of summer 1, professor is lazily going through register flow in April) 

Could we do something where we match the current day by the lowest difference in number of days from the "start dates" of each semester? Is that too much?

ex: Today is April 26, 2022.
`Spring_startDate`: January 18, 2022
`Summer_1_startDate`: May 1, 2022
today - summer_1 = 4 days
today - spring = 100+ days
...therefore set today as `Summer_1` Semester.

Reply:
I added a proof of concept in this [sandbox](https://replit.com/@nicoledanuwidja/FailingSlategrayNotifications#index.js)

Reply:
This is really good!! I think it will do the job.